src/transformers/generation_utils.py

```diff
@@ -1103,6 +1132,9 @@ def greedy_search(
                 An instance of :class:`~transformers.LogitsProcessorList`. List of instances of class derived from
                 :class:`~transformers.LogitsProcessor` used to modify the prediction scores of the language modeling
                 head applied at each generation step.
+            stopping_criteria (:obj:`StoppingCriteriaList`, `optional`):
+                An instance of :class:`~transformers.StoppingCriteriaList`. List of instances of class derived from
+                :class:`~transformers.StoppingCriteria` used to tell if the generation loop should stop.
             max_length (:obj:`int`, `optional`, defaults to 20):
```

Comment:
Can we remove `max_length` from the doctsrings and deprecate it?

Reply:
I can't because of this:

https://github.com/huggingface/transformers/blob/master/src/transformers/generation_utils.py#L1418

I'm not sure how to solve this. This probably leads to a significant perf loss if we remove it (not reallocating is important IMO, but that as usual would need some actual testing to make sure.).

In any case, I felt that this PR was already large enough as is, and that deprecating `max_length` and handling its usage (here and in `beam_sample` and `beam_search`) is something a bit more involved and would require some discussions which should not prevent (IMHO) to pass this PR

Reply:
One thing we have to do in the PR before merging IMO is to make sure we ensure full backward compatibility though. If I understand correctly, if one directly uses `greedy_search` at the moment instead of calling `generate()`, then `stopping_criteria` has to be provided or else the generation would not stop. 

Could we for now check if a `MaxLengthCriteria` is in the list atnd if yes and if it's max_length is != the given provided max_length => overwrite it? And if no `MaxLengthCriteria` is provided, we'll just add it?

Reply:
It does.